### ISSUE TITLE
Increate the count only when creation success.

### ISSUE THREAD

**Issue by I322871** [ramiel/mongoose-sequence#99]
Dear Experts,

I found the count is increased no matter if the creation of the document success or not. E.g. 

1. I create a new user, the count will be 1
2. I try to create the second user, but since I give a duplicate key to it, the creation failed (err.name === "MongoError" && err.code === 11000)
3. I create a new user, the count is 3, but actually it should be 2.

Any idea on this? Thank you so much!

**Comment by ramiel** [comment]
Yes the hook used is a pre-save hook, so the counter is incremented before the document is actually saved. This is safer than a post hook and better mimic the SQL behavior. At the moment there's no option to use a post hook but you can use a workaround. You can disable hooks completely and run manual increment by yourself. See the [disable_hooks options](https://github.com/ramiel/mongoose-sequence#options) and the [manual increment documentation](https://github.com/ramiel/mongoose-sequence#not-automatic-sequences)
